extension/opampextension/config_test.go

```diff
@@ -41,20 +41,141 @@ func TestUnmarshalConfig(t *testing.T) {
 		}, cfg)
 }
 
-func TestConfigValidate(t *testing.T) {
-	cfg := &Config{
-		Server: &OpAMPServer{
-			WS: &OpAMPWebsocket{},
+func TestUnmarshalHttpConfig(t *testing.T) {
+	cm, err := confmaptest.LoadConf(filepath.Join("testdata", "config_http.yaml"))
+	require.NoError(t, err)
+	factory := NewFactory()
+	cfg := factory.CreateDefaultConfig()
+	assert.NoError(t, component.UnmarshalConfig(cm, cfg))
+	assert.Equal(t,
+		&Config{
+			Server: &OpAMPServer{
+				HTTP: &commonFields{
+					Endpoint: "https://127.0.0.1:4320/v1/opamp",
+				},
+			},
+			InstanceUID: "01BX5ZZKBKACTAV9WEVGEMMVRZ",
+			Capabilities: Capabilities{
+				ReportsEffectiveConfig: true,
+			},
+		}, cfg)
+}
+
+func TestConfig_Validate(t *testing.T) {
+	type fields struct {
+		Server       *OpAMPServer
+		InstanceUID  string
+		Capabilities Capabilities
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		wantErr assert.ErrorAssertionFunc
+	}{
+		{
+			name: "WS must have endpoint",
+			fields: fields{
+				Server: &OpAMPServer{
+					WS: &commonFields{},
+				},
+			},
+			wantErr: func(t assert.TestingT, err error, i ...any) bool {
```

Comment:
```suggestion
			wantErr: func(t assert.TestingT, err error, _ ...any) bool {
```
I don't think these are used, right?

Reply:
boilerplate from IDE 🙃